internal/args/args.go

```diff
@@ -24,13 +24,17 @@ import (
-	QoSBaseConfigPresent     = flag.Bool("arg_qos_baseconfig_present", true, "QoS Counter subtest in gNMI-1.10 requires related base config to be loaded. Use this flag to skip the when base config is not loaded.")
-	LACPBaseConfigPresent    = flag.Bool("arg_lacp_baseconfig_present", true, "LACP subtest in gNMI-1.10 requires related base config to be loaded. Use this flag to skip the test when base config is not loaded.")
+	NumControllerCards      = flag.Int("arg_num_controller_cards", -1, "The expected number of controller cards. Some devices with a single controller report 0, which is a valid expected value. Expectation is not checked for values < 0.")
+	NumLinecards            = flag.Int("arg_num_linecards", -1, "The expected number of linecards. Some devices with a single linecard report 0, which is a valid expected value. Expectation is not checked for values < 0.")
+	P4RTNodeName1           = flag.String("arg_p4rt_node_name_1", "", "The P4RT Node Name for the first FAP. Test that reserves ports in the same FAP should configure this P4RT Node. The value will only be used if deviation ExplicitP4RTNodeComponent is applied.")
+	P4RTNodeName2           = flag.String("arg_p4rt_node_name_2", "", "The P4RT Node Name for the second FAP. Test that reserves ports in two different FAPs should configure this P4RT Node in addition to the Node defined in P4RTNodeName1. The value will only be used if deviation ExplicitP4RTNodeComponent is applied.")
+	FullConfigReplaceTime   = flag.Duration("arg_full_config_replace_time", 0, "Time taken for gNMI set operation to complete full configuration replace. Expected duration is in nanoseconds. Expectation is not checked when value is 0.")
+	SubsetConfigReplaceTime = flag.Duration("arg_subset_config_replace_time", 0, "Time taken for gNMI set operation to modify a subset of configuration. Expected duration is in nanoseconds. Expectation is not checked when value is 0.")
+	QoSBaseConfigPresent    = flag.Bool("arg_qos_baseconfig_present", true, "QoS Counter subtest in gNMI-1.10 requires related base config to be loaded. Use this flag to skip the when base config is not loaded.")
+	LACPBaseConfigPresent   = flag.Bool("arg_lacp_baseconfig_present", true, "LACP subtest in gNMI-1.10 requires related base config to be loaded. Use this flag to skip the test when base config is not loaded.")
+	TempSensorNamePattern   = flag.String("arg_temp_sensor_name_pattern", "^[0-9]/RP[0-9]/CPU[0-9].*TEMP.*", "There is no component type specifically for temperature sensors. So, we use the name pattern to find them.")
+	SwitchChipNamePattern   = flag.String("arg_switchchip_name_pattern", "^[0-9]/[0-9]/CPU[0-9]-NPU[0-9]", "There is no component type specifically for SwitchChip components. So, we use the name pattern to find them.")
```

Comment:
Would it make sense to use a switch statement on dut.vendor() instead of an argument?
Also, we shouldn't be using vendor-specific defaults for common arguments IMO.

Reply:
addressed your comment about the default values. missed to clean out when pushed the code.
with switch each vendor should open a separate pr. this way we can all use it. Also, not all vendors require patterns for all components type. That is why i used args that eventually can be removed when OC component model is improved. I can still open to make the change if you have prefer that way.